### ISSUE TITLE
Idle system having high load average

### ISSUE THREAD

**Issue by abasu0713** [orangepi-xunlong/linux-orangepi#76]
Hi OrangePi team,

I am compiling a full Linux Image and loading it on a board with some extra Kernel Modules (Ceph and RBD).
It's fresh install and has no software installed on it. Not even an update. Can you guys tell me why am I seeing a load average of 1.0 and higher all the time? and there are hundreds of processes for zswap. 

```bash
orangepi@orangepi3b:~$ uptime
 00:58:32 up 47 min,  3 users,  load average: 1.02, 1.01, 1.00
```

<img width="932" alt="Screenshot 2024-05-09 at 7 57 12 PM" src="https://github.com/orangepi-xunlong/linux-orangepi/assets/11951136/fbaf7dab-8dd8-4c42-84fb-d3b999f20d8d">

![image](https://github.com/orangepi-xunlong/linux-orangepi/assets/11951136/6e7735db-b440-4ed6-a238-3dcddfcd6d70)

I don't have this issue on an OrangePi 5B or 5Plus with the same kernel modules self-compiled. 

<img width="744" alt="Screenshot 2024-05-09 at 8 00 46 PM" src="https://github.com/orangepi-xunlong/linux-orangepi/assets/11951136/d1ed7134-da45-483e-9b3c-569d77a10946">

<img width="901" alt="Screenshot 2024-05-09 at 8 00 14 PM" src="https://github.com/orangepi-xunlong/linux-orangepi/assets/11951136/de9ba498-36e3-44ad-88c4-113fdbf94850">

It's a simple Ubuntu Server Jammy. 

```bash
orangepi@orangepi3b:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.4 LTS
Release:	22.04
Codename:	jammy
orangepi@orangepi3b:~$ uptime
 01:02:30 up 51 min,  3 users,  load average: 1.07, 1.02, 1.00
```

I have even tried turning off swap. To no avail. Would really appreciate any help

**Comment by ValdikSS** [comment]
That's because some of the custom driver/kernel modifications use kernel threads which are treated as 'not idle' by loadavg.
Usually this happens when `wait_event()` is used, which flags the thread as TASK_UNINTERRUPTIBLE, which is counted as 'busy' by loadavg. The CPU is not under load with this code style, just an incorrect loadavg reporting.

It could be `sdiohal_tx_thread` which you see in top.

**Comment by abasu0713** [comment]
> That's because some of the custom driver/kernel modifications use kernel threads which are treated as 'not idle' by loadavg.
> Usually this happens when `wait_event()` is used, which flags the thread as TASK_UNINTERRUPTIBLE, which is counted as 'busy' by loadavg. The CPU is not under load with this code style, just an incorrect loadavg reporting.
> 
> It could be `sdiohal_tx_thread` which you see in top.

Yeah. I couldn't find the right reasons. I moved to Debian server and used armbian. And have no such issues there. Been super stable as well.

**Comment by baiywt** [comment]
We've tested with Orangepi3b_1.0.8_ubuntu_jammy_desktop_xfce_linux5.10.160.img and the load average appears to be normal on our side.

![Image](https://github.com/user-attachments/assets/c039bde0-20f9-4a50-af89-841c9cf18692)
